### PR TITLE
feat(IAM Identity):Trusted Profile Identity api

### PR DESCRIPTION
**Pull request by Somchavan**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
=== RUN   TestAccIBMIamTrustedProfileIdentitiesDataSourceBasic
--- PASS: TestAccIBMIamTrustedProfileIdentitiesDataSourceBasic (61.48s)
PASS
...
=== RUN   TestAccIBMIamTrustedProfileIdentityDataSourceBasic
--- PASS: TestAccIBMIamTrustedProfileIdentityDataSourceBasic (31.60s)
PASS
...
=== RUN   TestAccIBMIamTrustedProfileIdentityAllArgs
--- PASS: TestAccIBMIamTrustedProfileIdentityAllArgs (65.76s)
PASS
...
=== RUN   TestAccIBMIamTrustedProfileIdentityBasic
--- PASS: TestAccIBMIamTrustedProfileIdentityBasic (33.85s)
```

### CODE REVIEW

File: ibm/service/iamidentity/resource_ibm_iam_trusted_profile_identity.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+				Description:  "Type of the identity.",
+			},
+			"accounts": &schema.Schema{
+				Type:        schema.TypeList,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Only valid for the type user. Accounts from which a user can assume the trusted profile.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"description": &schema.Schema{
```
Comment (hkantare) [inline ibm/service/iamidentity/resource_ibm_iam_trusted_profile_identity.go:62]: change in description is force_new (which destroy and recreate the resource)

File: website/docs/d/iam_trusted_profile_identities.html.markdown [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+data "ibm_iam_trusted_profile_identities" "iam_trusted_profile_identities" {
+	profile_id = "profile_id"
+}
+```
+
+## Argument Reference
+
+Review the argument reference that you can specify for your data source.
+
+* `profile_id` - (Required, Forces new resource, String) ID of the trusted profile.
```
Comment (hkantare) [inline website/docs/d/iam_trusted_profile_identities.html.markdown:25]: For datasources "Forces new resource" is not valid

File: website/docs/d/iam_trusted_profile_identities.html.markdown [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+# ibm_iam_trusted_profile_identities
+
+Provides a read-only data source for iam_trusted_profile_identities. You can then reference the fields of the data source in other resources within the same configuration using interpolation syntax.
+
+## Example Usage
+
+```hcl
+data "ibm_iam_trusted_profile_identities" "iam_trusted_profile_identities" {
+	profile_id = "profile_id"
```
Comment (hkantare) [inline website/docs/d/iam_trusted_profile_identities.html.markdown:17]: Can we provide some valid dummy values or can we use some interpolation instead to fetch the "profile_id" from other resource/datasource
Holds for other doc files also

File: ibm/service/iamidentity/data_source_ibm_iam_trusted_profile_identity.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "ID of the trusted profile.",
+			},
+			"identity_type": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Type of the identity.",
+			},
+			"identifier_id": &schema.Schema{
```
Comment (hkantare) [inline ibm/service/iamidentity/data_source_ibm_iam_trusted_profile_identity.go:34]: can we have common naming convention for "identifier_id" . we are refeering "identifier_id" and "identifier" in different places

File: website/docs/r/iam_trusted_profile_identity.html.markdown [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  * Constraints: Allowable values are: `user`, `serviceid`, `crn`.
+* `profile_id` - (Required, Forces new resource, String) ID of the trusted profile.
+* `type` - (Required, Forces new resource, String) Type of the identity.
+  * Constraints: Allowable values are: `user`, `serviceid`, `crn`.
+
+## Attribute Reference
+
+In addition to all argument references listed, you can access the following attribute references after your resource is created.
+
+* `id` - The unique identifier of the iam_trusted_profile_identity.
```
Comment (hkantare) [inline website/docs/r/iam_trusted_profile_identity.html.markdown:41]: update the description its combination of multiple values
```
<profile-id>/<identity-type>/<identifier-id>
```

File: ibm/service/iamidentity/resource_ibm_iam_trusted_profile_identity.go (hunk 1 above)
Comment (Somchavan) [reply ibm/service/iamidentity/resource_ibm_iam_trusted_profile_identity.go:62]: accommodated have been accomodated.

File: website/docs/d/iam_trusted_profile_identities.html.markdown (hunk 2 above)
Comment (Somchavan) [reply website/docs/d/iam_trusted_profile_identities.html.markdown]: changes have been accomodated.

File: website/docs/d/iam_trusted_profile_identities.html.markdown (hunk 3 above)
Comment (Somchavan) [reply website/docs/d/iam_trusted_profile_identities.html.markdown]: changes have been accomodated.

File: ibm/service/iamidentity/data_source_ibm_iam_trusted_profile_identity.go (hunk 4 above)
Comment (Somchavan) [reply ibm/service/iamidentity/data_source_ibm_iam_trusted_profile_identity.go:34]: identifier_id and identifier are two different variables. they are being refered according to the context.

File: website/docs/r/iam_trusted_profile_identity.html.markdown (hunk 5 above)
Comment (Somchavan) [reply website/docs/r/iam_trusted_profile_identity.html.markdown:41]: removed id part as it is not a part for output. changes have been accommodated.

File: ibm/service/iamidentity/resource_ibm_iam_trusted_profile_identity.go (hunk 1 above)
Comment (hkantare) [reply ibm/service/iamidentity/resource_ibm_iam_trusted_profile_identity.go:62]: What changes were accomodated we still see 
ForceNew:    true,?

File: website/docs/r/iam_trusted_profile_identity.html.markdown (hunk 5 above)
Comment (hkantare) [reply website/docs/r/iam_trusted_profile_identity.html.markdown:41]: by default Terraform add "id" as part of each res/ds so it will be always in output

**Pull request merged**